From 64d2a80d2de94bbf3b94f2e3307cf654a48ac6ba Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Fri, 2 Nov 2018 14:10:57 +0800
Subject: [PATCH 4998/5242] MLK-20177-3 usb: chipidea: imx: disable stream
 mode for imx8mm

commit  3b41f3a86a130082ee7799353c03c21aa09ec11d from
https://source.codeaurora.org/external/imx/linux-imx.git

Add CI_HDRC_DISABLE_DEVICE_STREAMING flag for iMX8mm usb device mode.

Acked-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/chipidea/ci_hdrc_imx.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/usb/chipidea/ci_hdrc_imx.c b/drivers/usb/chipidea/ci_hdrc_imx.c
index 845746c..e3a7fdd 100644
--- a/drivers/usb/chipidea/ci_hdrc_imx.c
+++ b/drivers/usb/chipidea/ci_hdrc_imx.c
@@ -86,6 +86,11 @@ struct ci_hdrc_imx_platform_flag {
 	.flags = CI_HDRC_SUPPORTS_RUNTIME_PM,
 };
 
+static const struct ci_hdrc_imx_platform_flag imx8mm_usb_data = {
+	.flags = CI_HDRC_SUPPORTS_RUNTIME_PM |
+		CI_HDRC_DISABLE_DEVICE_STREAMING,
+};
+
 static const struct of_device_id ci_hdrc_imx_dt_ids[] = {
 	{ .compatible = "fsl,imx23-usb", .data = &imx23_usb_data},
 	{ .compatible = "fsl,imx28-usb", .data = &imx28_usb_data},
@@ -97,6 +102,7 @@ struct ci_hdrc_imx_platform_flag {
 	{ .compatible = "fsl,imx7d-usb", .data = &imx7d_usb_data},
 	{ .compatible = "fsl,imx7ulp-usb", .data = &imx7ulp_usb_data},
 	{ .compatible = "fsl,imx8qm-usb", .data = &imx8qm_usb_data},
+	{ .compatible = "fsl,imx8mm-usb", .data = &imx8mm_usb_data},
 	{ /* sentinel */ }
 };
 MODULE_DEVICE_TABLE(of, ci_hdrc_imx_dt_ids);
-- 
1.7.9.5

